src/util.js

```diff
@@ -406,16 +406,28 @@
    */
   connect.PopupManager = function () { };
 
-  connect.PopupManager.prototype.open = function(url, name) {
+  connect.PopupManager.prototype.open = function(url, name, forceWindow) {
     var then = this._getLastOpenedTimestamp(name);
     var now = new Date().getTime();
     var win = null;      
     if (now - then > ONE_DAY_MILLIS) {
-       win = window.open('', name);
-       if (win.location !== url) {
+      if (forceWindow === true) {
```

Comment:
This appears to be a very specific use case. Why not just allow [`windowFeatures`](https://developer.mozilla.org/en-US/docs/Web/API/Window/open) to be passed in instead?

Reply:
These are the standard features I'd expect anyone trying to pop a standard login pop-up to include. Requiring the knowledge and overhead to include them all just to get a traditional login pop-up feels like a lot of avoidable burden to put on users - especially since centering the popup on the screen requires logic, not just a setting.

Reply:
My personal preference would be to allow `windowFeatures` as you described, but make pre-defined pop-up settings like this the default. 

While I understand deferring to the default browser settings, I imagine not very many users will actually want to use them, since most browsers default to a new tab. 

If they do want a new tab, they can simply overwrite the `windowFeatures` with an empty string, a much easier change than adding all this logic every time you want a traditional user login experience.

Reply:
I can see either side tho - tell me what will get it through and I'll make the change. :)

Reply:
@dinkzilla I have discussed this with my team. We would like to be able to group similar configurations together (similar to `softphone` property), so ideally we would not have both `loginPopup` and `loginPopupForceWindow`.

It would be good to group everything under `loginPopup`, something like this:

```ts
interface {
  loginPopup?: boolean | {
    width?: number; // with some default value (can you comment on why 573 and 400 are chosen)
    height?: number; // same as above
    forceWindow?: boolean // default to false
  }
}
```

where `loginPopup` is truthy (e.g. empty object `{}` should also evaluate to `true`) and defaults to `true`.

For `connect.PopupManager.prototype.open`, I think you can just add a new options parameter instead of `forceWindow` for extensibility reasons, something like:

```ts
open(url: string, name: string, options?: { forceWindow?: boolean,  width?: number, height?: number} = {})
```

Per https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Window_features, 

> If the strWindowFeatures parameter is used, the features that are not listed will be disabled or removed (except titlebar and close, which are by default yes).

The `toolbar=no, location=no, directories=no, status=no, menubar=no, scrollbars=no, resizable=no, copyhistory=no` portion is not needed.

By the way, I am curious how this deals with popup blockers. From a quick test I've performed on my end it seems like the new window will be blocked by the popup blocker, as the call is triggered asynchronously.

Reply:
@dinkzilla Are you still planning to implement this?

Reply:
Yeah I can do this. Sorry, had some time off. Will implment this week/weekend.

Reply:
Working on this now. To answer your question about pop-up blockers: In chrome, the new tab also gets blocked by the pop-up blocker and you have to allow it as well in our testing and I'm not sure of a way around it.

Reply:
I didn't put a lot of thought into the original height and width honestly, but after some examination, the connect login dialog is 355 x 490. It then has a bottom and top margin of 44. So I'm going with 433 x 578, which the minimum height that will not scroll and a width calculated to provide an even margin.

Reply:
So I made these changes. The problem now will be that this is a breaking change since `popupWindowAutoClose` used to be it's own object. I see three options:

Reply:
@haochi-aws any guidance on this?

Reply:
I think now that loginPopupAutoClose has been in the docs for a good while at this point, option 2 is our best bet.